packages/server/src/api/controllers/row/utils/utils.ts

```diff
@@ -117,6 +117,19 @@ export async function validate(
   })
 }
 
+function fixBooleanFields({ row, table }: { row: Row; table: Table }) {
+  for (let col of Object.values(table.schema)) {
+    if (col.type === FieldType.BOOLEAN) {
+      if (row[col.name] === 1) {
+        row[col.name] = true
+      } else if (row[col.name] === 0) {
+        row[col.name] = false
+      }
+    }
+  }
+  return row
+}
```

Comment:
SQS, MariaDB, and MySQL appear to store boolean values as numeric 1s or 0s, this function parses those back into proper booleans.